YCAI/src/state/popup.commands.ts

```diff
@@ -0,0 +1,28 @@
+import { HandshakeBody } from '@shared/models/HandshakeBody';
+import { command } from 'avenger';
+import * as TE from 'fp-ts/lib/TaskEither';
+import { Messages } from '../models/Messages';
+import { Settings } from '../models/Settings';
+import { API } from '../providers/api.provider';
+import { sendMessage } from '../providers/browser.provider';
+import { settings, settingsRefetch, keypair } from './popup.queries';
+
+export const handshake = command((handshake: HandshakeBody) =>
+  API.v3.Public.Handshake({ Body: handshake })
+);
+
+export const updateSettings = command(
+  (payload: Settings) => sendMessage(Messages.UpdateSettings)(payload),
+  { settings, settingsRefetch }
+);
+
+export const refreshSettings = command(() => TE.right(undefined), {
+  settings: settingsRefetch,
+});
+
+export const generateKeypair = command(
```

Comment:
since we're splitting the dashboard from the extension I think this should stay in the popup - and we may have an additional view that replicates what was inside `Settings`